src/esm.ts

```diff
@@ -28,7 +28,8 @@ export function registerAndCreateEsmHooks(opts?: RegisterOptions) {
     preferTsExts: tsNodeInstance.options.preferTsExts,
   });
 
-  return { resolve, getFormat, transformSource };
+  // return { resolve, getFormat, transformSource };
+  return { resolve, load };
```

Comment:
We will need to detect the version number of node and export hooks appropriate to that version, essentially supporting both the old and new loader APIs at the same time.  While it's tempting to say that we break backwards-compatibility, this is really inconvenient in practice.  There are many reasons a user might be on node 14 -- it is the LTS version -- and I don't want to bar those users from access to newer ts-node features.

Reply:
Yes, I am actually using v14 myself with ts-node/esm so I agree backwards compat is necessary. I will look into you example above.

Reply:
Since `resolve` seem unchanged and load is new, I think we can just return an object with all hooks and that would make it backwards compatible. I'll give that a try.

Reply:
I'm now checking the version to avoid the deprecation warning.